monkey/monkey_island/cc/setup/island_event_handlers.py

```diff
@@ -1,11 +1,34 @@
+from functools import partial
+
 from common import DIContainer
-from monkey_island.cc.event_queue import IslandEventTopic, PyPubSubIslandEventQueue
+from monkey_island.cc.event_queue import IIslandEventQueue, IslandEventTopic
 from monkey_island.cc.island_event_handlers import reset_agent_configuration
+from monkey_island.cc.repository import ICredentialsRepository
+from monkey_island.cc.services.database import Database
 
 
 def setup_island_event_handlers(container: DIContainer):
-    event_queue = container.resolve(PyPubSubIslandEventQueue)
+    island_event_queue = container.resolve(IIslandEventQueue)
+
+    _handle_reset_agent_configuration_events(island_event_queue, container)
+    _handle_clear_simulation_data_events(island_event_queue, container)
 
-    event_queue.subscribe(
+
+def _handle_reset_agent_configuration_events(
```

Comment:
`_handle` means that this method will "handle" something. It actually just subscribes to `IslandEventTopic.RESET_AGENT_CONFIGURATION`, the `reset_agent_configuration` is what handles the reset and `island_event_queue` handles the events. I'd suggest to use `_subscribe` prefix instead